sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -161,6 +164,46 @@ case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
   override def outputOrdering: Seq[SortOrder] = child.outputOrdering
 }
 
+/**
+ * Skip the first `offset` elements then take the first `limit` of the following elements in
+ *  the child's single output partition.
+ */
+case class GlobalLimitAndOffsetExec(
+    limit: Int,
+    offset: Int,
+    child: SparkPlan) extends BaseLimitExec {
+
+  override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
+
+  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
+
+  override def doExecute(): RDD[InternalRow] = {
+    val rdd = child.execute().mapPartitions { iter => iter.take(limit + offset)}
+    val skips = rdd.take(offset)
+    rdd.filter(!skips.contains(_))
```

Comment:
and here seems not taking duplicated rows into account. If it will filter out the all same rows selected by `offset`.

Reply:
Thank you! I will resolve this issue.